### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by knil-maloon**
I went ahead and created the pull request as this week is the last week they have budget for it and I would like to be able to add this on production.
There might be more changes coming but I think they won't be as big as this one. I wanted to make sure there was enough time to review. Better do the big one with time and any smaller ones later :)

- Added By Site graph:
     - New function and variable to get Institution Data.
- Fixed error on custom filters not showing hover.
- Arranged data in functions
- New constant for custom filters

### CODE REVIEW

**Review by mmcev106 — APPROVED** [review]
Approved!  The comments are all quite minor.

File: classes/CronData.php [hunk 1]
```diff
@@ -371,6 +371,7 @@ public static function getPercent($recordsTotal, $missing, $overall, $showLegend
 			$showLegend = true;
 		}else{
 			$percent = $overall;
+            $showLegend = true;
```
[the remark this answers is not in the document]
Comment (knil-maloon) [reply classes/CronData.php]: Just submitted the indentation fix. Figured no need to review this, heh

File: classes/GraphData.php [hunk 2]
```diff
@@ -234,27 +247,54 @@ public static function addGraphNoTops($graph,$question,$question_1,$study,$study
...
-            $graph = self::addGraph($graph,$question,"any",$study,$index,$survey_datetime);
+            $graph = self::addGraph($graph,$question,"any",$study,$studyCol,$survey_datetime,$institution);
         }
         return $graph;
     }
 
     public static function createPercentage($graph,$project_id,$study,$question,$question_1,$topScoreMax,$colType,$type,$date,$conditionDate,$recordIds){
         $condition = "";
-        if($colType != "total"){
+        if($colType != "total" && $study != "institutions"){
```
[the remark this answers is not in the document]
Comment (knil-maloon) [reply classes/GraphData.php:297]: That's a good idea. I might add it to might to do list so you might see it soon!

File: classes/R4Report.php [hunk 3]
```diff
@@ -81,6 +82,17 @@ public function getInstitutionData() {
 		}
 		return $this->institutionList;
 	}
+
+	public function setProjectDataInstitution($institution = ""){
+        if(!isset($this->projectDataAll)){
+            $this->projectDataAll = $this->getProjectData();
+        }
+	    if($institution !== ""){
```
[the remark this answers is not in the document]
Comment (knil-maloon) [reply classes/R4Report.php]: fixed

File: dashboard_cache.php [hunk 4]
```diff
@@ -226,7 +226,7 @@ function RGBToHSL(rgb) {
             if ($sstudy === $_SESSION[$project_id . "_study"]) {
                 $selected = "selected";
             }
-            if($customf_counter < 11) {
+            if($customf_counter < ProjectData::MAX_CUSTOM_FILTERS) {
```
[the remark this answers is not in the document]
Comment (knil-maloon) [reply dashboard_cache.php:229]: I'm trying to do more like this now that I have time and we need to use the budget. Trying to apply things from the Developer's meetings :)

**Pull request merged**